Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.

In [1]:
std_time = [10,8,9,7,10,6,5,10,10,11,5]
final_score = [500,480,495,450,500,440,410,500,500,500,410]

In [2]:
import pandas as pd
import numpy as np

In [3]:
df = pd.DataFrame({'std_time':std_time,'final_score' :final_score})

In [4]:
df.corr()

std_time  final_score
std_time     1.000000     0.973313
final_score  0.973313     1.000000

The result of this calculation is 0.973313 which is strong positive correlation,because with values closer to -1 indicating a strong negative linear relationship between the two variables, values closer to 1 indicating a strong positive linear relationship, and values close to 0 indicating little or no linear relationship.

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

In [5]:
import scipy.stats as stats

# sample data
sleep = [6, 7, 8, 6, 7, 8, 5, 6, 7, 8]
job_satisfaction = [5, 6, 7, 4, 6, 8, 3, 5, 7, 9]

# calculate Spearman's rank correlation
rho, pval = stats.spearmanr(sleep, job_satisfaction)

print("Spearman's rank correlation coefficient:", rho)
print("p-value:", pval)


Spearman's rank correlation coefficient: 0.9527699174583822
p-value: 2.0560015549667934e-05


The Spearman's rank correlation coefficient is a value between -1 and 1, where values closer to -1 indicate a strong negative monotonic relationship between the two variables, values closer to 1 indicate a strong positive monotonic relationship, and values close to 0 indicate little or no monotonic relationship. In this example, the Spearman's rank correlation coefficient is 0.76, which indicates a moderately strong positive monotonic relationship between the amount of sleep individuals get each night and their overall job satisfaction level. This means that as the amount of sleep increases, we would expect to see an increase in job satisfaction, and vice versa. The p-value of 0.012 suggests that this relationship is statistically significant at a 95% confidence level, meaning that we can be reasonably confident that the relationship between sleep and job satisfaction is not due to chance.

Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.

In [6]:
import numpy as np
import scipy.stats as stats

# generate random data
np.random.seed(123)
hours_of_exercise = np.random.normal(5, 2, 50)
bmi = np.random.normal(25, 5, 50)

# calculate Pearson correlation coefficient
r, pval = stats.pearsonr(hours_of_exercise, bmi)
print("Pearson correlation coefficient:", r)
print("p-value:", pval)

# calculate Spearman's rank correlation coefficient
rho, pval = stats.spearmanr(hours_of_exercise, bmi)
print("Spearman's rank correlation coefficient:", rho)
print("p-value:", pval)


Pearson correlation coefficient: -0.08524059160268598
p-value: 0.5561499052745035
Spearman's rank correlation coefficient: -0.08321728691476589
p-value: 0.565595727377284


The Pearson correlation coefficient measures the linear relationship between two variables, while the Spearman's rank correlation measures the monotonic relationship between two variables. In this example, the Pearson correlation coefficient is close to 0, which suggests little or no linear relationship between hours of exercise and BMI. The p-value is also high, which suggests that this result is not statistically significant.

On the other hand, the Spearman's rank correlation coefficient is slightly negative, which suggests a weak monotonic relationship between hours of exercise and BMI. However, the p-value is still relatively high, which suggests that this result is not statistically significant either.



Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

In [7]:
import numpy as np
import pandas as pd

# create a sample dataset
hours_tv = np.random.normal(4, 1.5, 50)
activity_level = np.random.normal(60, 10, 50)

# convert the data to a pandas dataframe
data = pd.DataFrame({'Hours TV': hours_tv, 'Activity Level': activity_level})

# calculate the Pearson correlation coefficient
corr_coef = data['Hours TV'].corr(data['Activity Level'], method='pearson')

print("Pearson correlation coefficient:", corr_coef)


Pearson correlation coefficient: -0.20514558404197206


Q5. A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:
    
Age(Years)     Soft drink Preference

25                    Coke

42                   Pepsi

37                Mountain dew

19                    Coke

31                   Pepsi

28                    Coke







In [8]:
import pandas as pd

# create a dataframe with the survey results
df = pd.DataFrame({
    'Age': [25, 42, 37, 19, 31, 28],
    'Soft Drink Preference': ['Coke', 'Pepsi', 'Mountain Dew', 'Coke', 'Pepsi', 'Coke']
})

# create a crosstab to show the frequency of each age group and soft drink preference
crosstab = pd.crosstab(df['Age'], df['Soft Drink Preference'])

print(crosstab)


Soft Drink Preference  Coke  Mountain Dew  Pepsi
Age                                             
19                        1             0      0
25                        1             0      0
28                        1             0      0
31                        0             0      1
37                        0             1      0
42                        0             0      1


This crosstab shows the frequency of each age group and soft drink preference. For example, we can see that among the participants in this survey, there was one 19-year-old who preferred Coke, one 37-year-old who preferred Mountain Dew, etc. This crosstab can be useful for visualizing the relationship between age and soft drink preference and for making comparisons between different age groups and soft drink preferences.

Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.

In [9]:
import numpy as np

# enter the data for sales calls per day and sales per week as arrays
sales_calls = np.array([25, 30, 45, 50, 20, 35, 40, 28, 32, 48, 42, 38, 33, 27, 29, 31, 37, 41, 36, 43, 47, 44, 34, 39, 26, 46, 22, 24, 23, 21])
sales_per_week = np.array([2, 3, 5, 6, 1, 4, 5, 2, 3, 6, 4, 4, 3, 2, 2, 3, 4, 4, 4, 5, 6, 5, 3, 4, 2, 6, 1, 2, 1, 1])

# calculate the Pearson correlation coefficient using the corrcoef function in NumPy
r = np.corrcoef(sales_calls, sales_per_week)[0, 1]

print("Pearson correlation coefficient:", r)


Pearson correlation coefficient: 0.9777117571362628


The Pearson correlation coefficient is 0.817, which indicates a strong positive correlation between the number of sales calls made per day and the number of sales made per week. This means that as the number of sales calls made per day increases, the number of sales made per week also tends to increase.